docs/develop/porting.rst

```diff
@@ -0,0 +1,325 @@
+.. _porting_to_a_board:
+
+Porting MicroPython
+====================
+
+The MicroPython project contains several ports to different microcontroller families and
+architectures. The project repository has a `ports <https://github.com/micropython/micropython/tree/master/ports>`_
+directory containing a subdirectory for each supported port.
+
+A port will typically contain definitions for multiple "boards", each of which is a specific piece of
+hardware that that port can run on (i.e. a development kit or device).
+
+The
+`minimal port <https://github.com/micropython/micropython/tree/master/ports/minimal>`_ is 
+available as a simplified reference implementation of a MicroPython port. 
+In general, starting a port requires:
+
+- Setting up the toolchain (configuring Makefiles, etc)
+- Implementing boot configuration and CPU initialization
+- Initialising basic drivers required for development and debugging (e.g. GPIO, UART).
+- Performing the board-specific configurations.
+- Implementing the port-specific modules.
+
+
+The minimal MicroPython firmware
+---------------------------------
+
+The start of porting MicroPython to a new board is by integrating the minimal firmware.
+For this walkthrough, create a subdirectory for the new port in the ``ports`` directory:
+
+.. code-block:: bash
+
+   cd ports
+   mkdir example_port
+
+The minimal MicroPython firmware specifications are implemented in the main port file i.e ``main.c``:
+
+.. code-block:: c
+
+    #include "py/compile.h"
+    #include "py/runtime.h"
+    #include "py/repl.h"
+    #include "py/gc.h"
+    #include "lib/utils/pyexec.h"
+
+    #include <stdio.h>
+
+    static char heap[2048];
+
+    int main (int argc, char **argv) {
+    #if MICROPY_ENABLE_GC
+        gc_init(heap, heap + sizeof(heap));
+    #endif
```

Comment:
for this walkthrough it might be simpler to leave out these #if's, ie just have gc_init() unconditionally